// @netlify/sdk--ui-core/dist/message_schema/surface/request_dimensions.js
/*
 * The REQUEST_DIMENSIONS message is sent when the surface would like the host to resize its
 * iframe element.
 *
 * Because the surface is rendered into a sandboxed iframe, the host cannot observe the dimensions
 * of the frame's body and must rely on the surface to report its desired dimensions. (In general,
 * the surface will simply ask for the dimensions of its body by setting up a ResizeObserver and
 * triggering a message on resize events.)
 */
import { z } from "zod";
export const UI_EXTENSION_SURFACE_REQUEST_DIMENSIONS_EVENT_TYPE = "NF::EUI_SURFACE::REQUEST_DIMENSIONS";
export const UIExtensionSurfaceResizeEventSchema = z.object({
    type: z.literal(UI_EXTENSION_SURFACE_REQUEST_DIMENSIONS_EVENT_TYPE),
    data: z.object({
        dimensions: z.object({
            height: z.number().positive(),
            width: z.number().positive(),
        }),
    }),
});
export const createUIExtensionSurfaceResizeEvent = (data) => ({
    type: UI_EXTENSION_SURFACE_REQUEST_DIMENSIONS_EVENT_TYPE,
    data,
});
