// @netlify/sdk--ui-react/dist/components/Form/Fields/Select.js
import { jsx } from "react/jsx-runtime";
import "react";
import { u as useFormContext, C as Controller } from "../../../index.esm-BuArIOt4.js";
import FormField from "../../vendor/FormField.js";
import Select$1 from "../../vendor/Select/Select.js";
const Select = ({
  label,
  helpText,
  name,
  options,
  required,
  disabled,
  error,
  clearable: isClearable,
  ...otherProps
}) => {
  const { control } = useFormContext();
  return /* @__PURE__ */ jsx(
    Controller,
    {
      name,
      control,
      render: ({ field, fieldState, formState }) => {
        var _a;
        return /* @__PURE__ */ jsx(
          FormField,
          {
            disabled,
            error: (error == null ? void 0 : error.message) ?? ((_a = fieldState.error) == null ? void 0 : _a.message),
            helpText,
            label,
            control: /* @__PURE__ */ jsx(
              Select$1,
              {
                ...field,
                valueOnly: true,
                required,
                options,
                isClearable,
                isDisabled: formState.disabled || disabled,
                ...otherProps
              }
            )
          }
        );
      }
    }
  );
};
export {
  Select
};
