// @netlify/sdk--ui-react/dist/components/vendor/Alert/Alert.js
import { jsx, jsxs } from "react/jsx-runtime";
import { cn, DecorativeIcon, ButtonIcon } from "@netlify/ui";
import { parseProps } from "./helpers.js";
import { alertCN, iconCN, contentsCN, dismissButtonCN } from "./styles.js";
const AlertContents = ({
  as: Component = "div",
  id,
  children,
  className
}) => {
  return /* @__PURE__ */ jsx(Component, { id, className: cn("alert-markdown", className), children });
};
const Alert = ({
  title,
  onDismiss,
  outset,
  className,
  iconClassName,
  ...rest
}) => {
  const { message, role, state, icon } = parseProps(rest);
  if (!message) {
    return null;
  }
  return /* @__PURE__ */ jsxs(
    "div",
    {
      id: rest.id,
      className: alertCN({ outset, state, className }),
      role,
      children: [
        /* @__PURE__ */ jsx(
          DecorativeIcon,
          {
            name: icon,
            className: iconCN({ state, iconClassName })
          }
        ),
        /* @__PURE__ */ jsx(AlertContents, { className: contentsCN, children: message }),
        onDismiss && /* @__PURE__ */ jsx(
          ButtonIcon,
          {
            iconWrapperClassName: "tw-right-0",
            level: "tertiary",
            onClick: () => {
              onDismiss();
            },
            className: dismissButtonCN,
            width: 24,
            iconName: "close",
            title: ["Close", title].filter(Boolean).join(": ")
          }
        )
      ]
    }
  );
};
export {
  AlertContents,
  Alert as default
};
