// @netlify/netlify-design-tokens/dist/js/index.js
export const colorFacetsNeutralLight100 = "#f6f6f7";
export const colorFacetsNeutralLight200 = "#e9ebed";
export const colorFacetsNeutralLight300 = "#d1d5da";
export const colorFacetsNeutralLight400 = "#9da7b2";
export const colorFacetsNeutralLight500 = "#778089";
export const colorFacetsNeutralLight600 = "#545a61";
export const colorFacetsNeutralLight700 = "#353a3e";
export const colorFacetsNeutralLight800 = "#181a1c";
export const colorFacetsNeutralLight000 = "#ffffff";
export const colorFacetsNeutralDark100 = "#e6ecf2";
export const colorFacetsNeutralDark200 = "#abb5bf";
export const colorFacetsNeutralDark300 = "#7e8792";
export const colorFacetsNeutralDark400 = "#4d565f";
export const colorFacetsNeutralDark500 = "#3b434c";
export const colorFacetsNeutralDark600 = "#272f38";
export const colorFacetsNeutralDark700 = "#1e242c";
export const colorFacetsNeutralDark800 = "#12181f";
export const colorFacetsNeutralDark900 = "#060b10";
export const colorFacetsNeutralDark000 = "#ffffff";
export const colorFacetsTeal100 = "#8efbf7";
export const colorFacetsTeal200 = "#32e6e2";
export const colorFacetsTeal300 = "#14d8d4";
export const colorFacetsTeal400 = "#05bdba";
export const colorFacetsTeal500 = "#04a29f";
export const colorFacetsTeal600 = "#02807d";
export const colorFacetsTeal700 = "#016968";
export const colorFacetsTeal800 = "#014847";
export const colorFacetsTeal900 = "#0c2a2a";
export const colorFacetsTeal000 = "#defffe";
export const colorFacetsGreen100 = "#bef9c6";
export const colorFacetsGreen200 = "#93f5a5";
export const colorFacetsGreen300 = "#64d87f";
export const colorFacetsGreen400 = "#3ac364";
export const colorFacetsGreen500 = "#31a855";
export const colorFacetsGreen600 = "#228240";
export const colorFacetsGreen700 = "#1a6b34";
export const colorFacetsGreen800 = "#0f4a21";
export const colorFacetsGreen900 = "#152a19";
export const colorFacetsGreen000 = "#e7fce9";
export const colorFacetsGold100 = "#f6e0a5";
export const colorFacetsGold200 = "#facd6f";
export const colorFacetsGold300 = "#fbb13d";
export const colorFacetsGold400 = "#f98e21";
export const colorFacetsGold500 = "#d5771a";
export const colorFacetsGold600 = "#a85d13";
export const colorFacetsGold700 = "#8c4c0d";
export const colorFacetsGold800 = "#603408";
export const colorFacetsGold900 = "#332213";
export const colorFacetsGold000 = "#fdf5d8";
export const colorFacetsRed100 = "#fbd3d0";
export const colorFacetsRed200 = "#ffbdba";
export const colorFacetsRed300 = "#ffada9";
export const colorFacetsRed400 = "#fe8382";
export const colorFacetsRed500 = "#fe4e5c";
export const colorFacetsRed600 = "#d62740";
export const colorFacetsRed700 = "#af2536";
export const colorFacetsRed800 = "#800a20";
export const colorFacetsRed900 = "#3d1c1b";
export const colorFacetsRed000 = "#fbeeed";
export const colorFacetsPink100 = "#f6d2f2";
export const colorFacetsPink200 = "#f7bcf3";
export const colorFacetsPink300 = "#f3a8ee";
export const colorFacetsPink400 = "#ef7feb";
export const colorFacetsPink500 = "#df5adc";
export const colorFacetsPink600 = "#b938b8";
export const colorFacetsPink700 = "#9a2d99";
export const colorFacetsPink800 = "#6c1d6b";
export const colorFacetsPink900 = "#381b37";
export const colorFacetsPink000 = "#fcf0fb";
export const colorFacetsPurple100 = "#e2d9f7";
export const colorFacetsPurple200 = "#d8c7ff";
export const colorFacetsPurple300 = "#cab9f4";
export const colorFacetsPurple400 = "#b49df1";
export const colorFacetsPurple500 = "#9b80ed";
export const colorFacetsPurple600 = "#775ce7";
export const colorFacetsPurple700 = "#614aca";
export const colorFacetsPurple800 = "#382aa4";
export const colorFacetsPurple900 = "#292142";
export const colorFacetsPurple000 = "#f5f2fc";
export const colorFacetsBlue100 = "#cde2ff";
export const colorFacetsBlue200 = "#b5d2fb";
export const colorFacetsBlue300 = "#9cbef6";
export const colorFacetsBlue400 = "#80abfa";
export const colorFacetsBlue500 = "#5d8df5";
export const colorFacetsBlue600 = "#316bf4";
export const colorFacetsBlue700 = "#2e51ed";
export const colorFacetsBlue800 = "#2036a1";
export const colorFacetsBlue900 = "#1b205b";
export const colorFacetsBlue000 = "#edf4ff";
export const effectOpacity0 = 0;
export const effectOpacity10 = 0.1;
export const effectOpacity20 = 0.2;
export const effectOpacity38 = 0.38;
export const effectOpacity54 = 0.54;
export const effectOpacity100 = 1;
export const effectShadowLightShallow = "0 1px 10px 0 rgb(53 58 62 / 6%), 0 2px 4px 0 rgb(53 58 62 / 8%)";
export const effectShadowLightDeep = "0 16px 24px rgb(24 26 28 / 7%), 0 6px 30px rgb(24 26 28 / 6%), 0 8px 10px rgb(24 26 28 / 10%)"; // Use for modals, dropdowns, and other components that need to be elevated above the page.
export const effectShadowDarkShallow = "0 1px 10px 0 rgb(18 24 31 / 60%), 0 2px 4px 0 rgb(18 24 31 / 40%)";
export const effectShadowDarkDeep = "0 16px 24px rgb(6 11 16 / 20%), 0 6px 30px rgb(6 11 16 / 30%), 0 8px 10px rgb(6 11 16 / 40%)"; // Use for modals, dropdowns, and other components that need to be elevated above the page.
export const sizeBorderRadiusSmall = "6px";
export const sizeBorderRadiusBig = "8px";
export const typographySizeTiny = "12px";
export const typographySizeSmall = "14px";
export const typographySizeMedium = "16px";
export const typographySizeLarge = "18px";
export const typographySizeXl = "20px";
export const typographySizeXxl = "36px";
export const typographySizeXxxl = "64px";
export const typographySize4xl = "104px";
export const typographyWeightThin = 100;
export const typographyWeightLight = 300;
export const typographyWeightRegular = 400;
export const typographyWeightMedium = 500;
export const typographyWeightSemibold = 600;
export const typographyWeightBold = 700;
export const typographyWeightBlack = 900;
export const typographyLetterSpacingTight = "-0.025em";
export const typographyLetterSpacingNormal = 0;
export const typographyLetterSpacingWide = "1.5px";
export const typographyFontsSans = "system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol'";
export const typographyFontsMono = "ui-monospace, Menlo, Monaco, 'Cascadia Mono', 'Segoe UI Mono', 'Roboto Mono', 'Oxygen Mono', 'Ubuntu Monospace', 'Source Code Pro','Fira Mono', 'Droid Sans Mono', 'Courier New', monospace";
