// @netlify/sdk--ui-react/dist/components/FullstorySetup.js
import { jsx, Fragment } from "react/jsx-runtime";
import { i as initFS } from "../index.esm-B8tPjtbx.js";
import { useEffect } from "react";
var __assign = function() {
  __assign = Object.assign || function __assign2(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];
      for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
    }
    return t;
  };
  return __assign.apply(this, arguments);
};
var getFullStory = function() {
  if (window._fs_namespace) {
    return window[window._fs_namespace];
  }
  return void 0;
};
var _init = function(inputOptions, readyCallback) {
  var options = __assign({}, inputOptions);
  if (getFullStory()) {
    console.warn("The FullStory snippet has already been defined elsewhere (likely in the <head> element)");
    return;
  }
  if (options.recordCrossDomainIFrames) {
    window._fs_run_in_iframe = true;
  }
  if (options.appHost) {
    window._fs_app_host = options.appHost;
  }
  if (options.assetMapId) {
    window._fs_asset_map_id = options.assetMapId;
  }
  if (options.startCaptureManually) {
    window._fs_capture_on_startup = false;
  }
  if (options.recordOnlyThisIFrame) {
    window._fs_is_outer_script = true;
  }
  if (options.cookieDomain) {
    window._fs_cookie_domain = options.cookieDomain;
  }
  if (options.debug === true) {
    if (!options.script) {
      options.script = "edge.fullstory.com/s/fs-debug.js";
    } else {
      console.warn("Ignoring `debug = true` because `script` is set");
    }
  }
  initFS(options);
  var fs = getFullStory();
  if (!fs) {
    console.warn("Failed to initialize FS snippet");
    return;
  }
  if (readyCallback) {
    fs("observe", { type: "start", callback: readyCallback });
  }
  if (options.devMode === true) {
    var message = "FullStory was initialized in devMode and will stop recording";
    fs("trackEvent", {
      name: "FullStory Dev Mode",
      properties: {
        message
      }
    });
    fs("shutdown");
    window._fs_dev_mode = true;
    console.warn(message);
  }
};
var initOnce = function(message) {
  return function(inputOptions, readyCallback) {
    if (window._fs_initialized) {
      console.warn(message);
      return;
    }
    _init(inputOptions, readyCallback);
    window._fs_initialized = true;
  };
};
var init = initOnce("FullStory init has already been called once, additional invocations are ignored");
const FullstorySetup = ({ children }) => {
  useEffect(() => {
    init({
      orgId: "o-1VMFTK-na1",
      namespace: "FS",
      recordCrossDomainIFrames: true
    });
  }, []);
  return /* @__PURE__ */ jsx(Fragment, { children });
};
export {
  FullstorySetup
};
