// @netlify/sdk--ui-react/dist/components/vendor/AlertSmall/styles.js
import { cn } from "@netlify/ui";
const STATES = {
  error: { className: "tw-text-red-darkest dark:tw-text-red-lightest" },
  warning: { className: "tw-text-gold-darkest dark:tw-text-gold-lighter" },
  info: {
    className: "tw-text-black dark:tw-text-gray-lightest",
    iconClassName: "tw-text-gray-dark dark:tw-text-gray-light"
  },
  success: { className: "tw-text-teal-darkest dark:tw-text-teal-lightest" }
};
const VARIANTS = {
  "form-field": "tw-mx-0.5 tw-text-sm tw-my-1 tw-leading-6 tw-font-regular"
};
const alertCN = ({
  outset,
  state,
  variant,
  className
}) => {
  var _a;
  return cn(
    "tw-relative tw-max-w-170 tw-pl-6 tw-text-left",
    outset,
    state && ((_a = STATES[state]) == null ? void 0 : _a.className),
    variant && VARIANTS[variant],
    className
  );
};
const iconCN = ({ state, iconClassName }) => {
  var _a;
  return cn("tw-absolute tw-left-0", state && ((_a = STATES[state]) == null ? void 0 : _a.iconClassName), iconClassName);
};
const contentsCN = "tw-c-alert-small tw-overflow-visible";
export {
  alertCN,
  contentsCN,
  iconCN
};
