// @netlify/sdk--ui-react/dist/use_netlify_extension_ui_fetch.js
import { useNetlifySDK } from "./use_netlify_sdk.js";
const createNetlifyExtensionUIFetch = ({
  fetch = globalThis.fetch,
  ...options
}) => {
  return (input, init) => {
    const headers = new Headers(init == null ? void 0 : init.headers);
    headers.set("Nf-UIExt-Extension-Id", options.extension.id);
    headers.set("Nf-UIExt-Extension-Slug", options.extension.slug);
    if (options.netlifyToken != null && options.netlifyToken !== "") {
      headers.set("Nf-UIExt-Netlify-Token", options.netlifyToken);
    }
    if (options.providerToken != null && options.providerToken !== "") {
      headers.set("Nf-UIExt-Provider-Token", options.providerToken);
    }
    headers.set(
      "Nf-UIExt-Provider-Tokens",
      JSON.stringify(options.providerTokens ?? [])
    );
    if (options.configurationId != null && options.configurationId !== "") {
      headers.set("Nf-UIExt-Configuration-Id", options.configurationId);
    }
    if (options.dataLayerId != null && options.dataLayerId !== "") {
      headers.set("Nf-UIExt-Datalayer-Id", options.dataLayerId);
    }
    if (options.siteId != null && options.siteId !== "") {
      headers.set("Nf-UIExt-Site-Id", options.siteId);
    }
    if (options.teamId != null && options.teamId !== "") {
      headers.set("Nf-UIExt-Team-Id", options.teamId);
    }
    if (options.user.id != null && options.user.id !== "") {
      headers.set("Nf-UIExt-User-Id", options.user.id);
    }
    return fetch(input, {
      ...init,
      headers
    });
  };
};
const useNetlifyExtensionUIFetch = () => {
  const sdk = useNetlifySDK();
  return (input, init) => {
    const fetch = createNetlifyExtensionUIFetch({
      configurationId: sdk.context.configurationId,
      dataLayerId: sdk.context.dataLayerId,
      extension: sdk.extension,
      netlifyToken: sdk.context.auth.netlifyToken,
      providerToken: sdk.context.auth.providerToken,
      providerTokens: sdk.context.auth.providerTokens,
      siteId: sdk.context.siteId,
      teamId: sdk.context.teamId,
      user: sdk.context.user
    });
    return fetch(input, init);
  };
};
export {
  useNetlifyExtensionUIFetch
};
