// wretch/dist/cjs/addons/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.progressAddon = exports.queryStringAddon = exports.perfsAddon = exports.formUrlAddon = exports.formDataAddon = exports.basicAuthAddon = exports.abortAddon = void 0;
var abort_js_1 = require("./abort.js");
Object.defineProperty(exports, "abortAddon", { enumerable: true, get: function () { return abort_js_1.default; } });
var basicAuth_js_1 = require("./basicAuth.js");
Object.defineProperty(exports, "basicAuthAddon", { enumerable: true, get: function () { return basicAuth_js_1.default; } });
var formData_js_1 = require("./formData.js");
Object.defineProperty(exports, "formDataAddon", { enumerable: true, get: function () { return formData_js_1.default; } });
var formUrl_js_1 = require("./formUrl.js");
Object.defineProperty(exports, "formUrlAddon", { enumerable: true, get: function () { return formUrl_js_1.default; } });
var perfs_js_1 = require("./perfs.js");
Object.defineProperty(exports, "perfsAddon", { enumerable: true, get: function () { return perfs_js_1.default; } });
var queryString_js_1 = require("./queryString.js");
Object.defineProperty(exports, "queryStringAddon", { enumerable: true, get: function () { return queryString_js_1.default; } });
var progress_js_1 = require("./progress.js");
Object.defineProperty(exports, "progressAddon", { enumerable: true, get: function () { return progress_js_1.default; } });
